src/MoBi.Core/Snapshots/Mappers/SimulationMapper.cs

```diff
@@ -89,6 +112,75 @@ public override async Task<Simulation> MapToSnapshot(MoBiSimulation simulation,
 
       snapshot.ParameterIdentificationWorkingDirectory = simulation.ParameterIdentificationWorkingDirectory;
 
+      snapshot.Parameters = await allParametersChangedByUserFrom(simulation);
+
+      snapshot.ScaleDivisors = allScaleDivisorsChangedByUserFrom(simulation);
+
       return snapshot;
    }
+
+   private void updateParameters(MoBiSimulation simulation, LocalizedParameter[] snapshotParameters)
+   {
+      snapshotParameters?.Each(x =>
+      {
+         var parameter = new ObjectPath(x.Path.ToPathArray()).TryResolve<IParameter>(simulation.Model.Root);
+         if (parameter != null)
+            changeQuantity(parameter, x, simulation);
+      });
+   }
+
+   private void updateScaleDivisors(MoBiSimulation simulation, ScaleDivisor[] snapshotScaleDivisors)
+   {
+      snapshotScaleDivisors?.Each(x =>
+      {
+         var moleculeAmount = new ObjectPath(x.Path.ToPathArray()).TryResolve<MoleculeAmount>(simulation.Model.Root);
+         if (moleculeAmount != null)
+            changeScaleDivisor(moleculeAmount, x, simulation);
+      });
+   }
+
+   private void changeScaleDivisor(MoleculeAmount moleculeAmount, ScaleDivisor scaleDivisor, MoBiSimulation simulation)
+   {
+      _quantityChangeTracker.TrackScaleChange(moleculeAmount, simulation, x => x.ScaleDivisor = scaleDivisor.Value, withEvents: false);
+   }
+
+   private void changeQuantity(IParameter parameter, LocalizedParameter snapshotParameter, MoBiSimulation simulation)
+   {
+      if (!snapshotParameter.Value.HasValue)
+         return;
+
+      _quantityChangeTracker.TrackQuantityChange(parameter, simulation, x =>
+      {
+         parameter.IsDefault = false;
+         x.Value = snapshotParameter.Value.Value;
+         x.DisplayUnit = _dimensionFactory.FindUnit(snapshotParameter.Unit).unit;
+         _valueOriginMapper.UpdateValueOrigin(x.ValueOrigin, snapshotParameter.ValueOrigin);
+      }, withEvents: false);
+   }
+
+   private ScaleDivisor[] allScaleDivisorsChangedByUserFrom(MoBiSimulation simulation)
+   {
+      var allAmountsToExport = new List<ScaleDivisor>();
+      simulation.OriginalQuantityValues.Where(x => x.Type == OriginalQuantityValue.Types.ScaleDivisor).Each(x =>
+      {
+         var moleculeAmount = new ObjectPath(x.Path.ToPathArray()).TryResolve<MoleculeAmount>(simulation.Model.Root);
+         if (moleculeAmount != null)
+            allAmountsToExport.Add(new ScaleDivisor { Path = x.Path, Value = moleculeAmount.ScaleDivisor });
+      });
+
+      return allAmountsToExport.ToArray();
+   }
+
+   private Task<LocalizedParameter[]> allParametersChangedByUserFrom(MoBiSimulation simulation)
+   {
+      var allParametersToExport = new List<IParameter>();
+      simulation.OriginalQuantityValues.Where(x => x.Type == OriginalQuantityValue.Types.Quantity).Each(x =>
+      {
+         var parameter = new ObjectPath(x.Path.ToPathArray()).TryResolve<IParameter>(simulation.Model.Root);
```

Comment:
lol yeah this is getting out of hand here. Let's create a method if it's not ther.e Iknow we have EntityAt. Maybe there is something similar?